core/server/middleware/cors.js

```diff
@@ -30,8 +31,27 @@ function getIPs() {
     return ips;
 }
 
-// origins that always match: localhost, local IPs, etc.
-whitelist = whitelist.concat(getIPs());
+function getUrls() {
+    var urls = [url.parse(config.url).hostname];
+
+    if (config.urlSSL) {
+        urls.push(url.parse(config.urlSSL).hostname);
+    }
```

Comment:
Do we ever need to add `urlSSL` to the whitelist? In the case where `urlSSL` is defined with an alternate URL, the admin client will be running on the same URL as the API so the requests will be same-origin. Unless I've overlooked something?